### DISCUSSION TITLE
Code error for Arduino Uno

### DISCUSSION THREAD

**Discussion by filut123** [revoxhere/duino-coin#840] [Q&A]
When loading the Duino coin on Arduino Uno I get this error "include sha1.h" no such file in directory. Do I need to upload some libraries first. Please advice. Thank you,

**Comment by Raabitco** [comment]
Hi, yes you need to update your library.
From (tools) choose (Manage Libraries), in the search write (JSON) and hit inter, after the results came out search in the list and find (Arduino Json) (by Binoit Blanchon) install it and everything will work fine, this is the solution that I am personally do and everything works perfect after that.
Hope this will help.

**Reply by filut123** [reply]
Thank you. I was able to find and upload 6.18.3 version by Binoit Blanchon but unfortunately still getting the same error. Any other advice. Thank you,

**Reply by filut123** [reply]
Arduino: 1.8.15 (Mac OS X), Board: "Arduino Uno"









Here is the full error shown when I copy Duino coin code into Arduino Uno and try to save it to Arduino:

sketch_aug23e:34:10: fatal error: sha1.h: No such file or directory
 #include "sha1.h"
          ^~~~~~~~
compilation terminated.
exit status 1
sha1.h: No such file or directory


This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

**Comment by Raabitco** [comment]
Really sorry, I do not have the information that can help to fix this issue, try to search online, maybe you could find some ideas.
for me will try to find as well, if I find something I will let you know.

**Reply by filut123** [reply]
Thank you,

**Comment by Raabitco** [comment]
Hi there, BTW, I buy ESP32 and when I tried to install the code I had the same error that you have LOL, So I do the details below and the arduino works perfectly.

Update your ESP32 core.
Open Arduino IDE > File > Preferences > In the Additional boards URL field put:
https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_dev_index.json

After doing that, update your ESP32 core: Arduino IDE > Tools > Board > Boards Manager > Type esp32 > Install version 2.0.0

You should now be able to compile the code without any problems.
The picture in attach show the success of the procedures and start mining.
![ESP32-success-connection](https://user-images.githubusercontent.com/79040691/131238259-c79f89ac-6896-4889-8980-891f3b6d0073.jpg)

**Reply by filut123** [reply]
I'm using cpu and not esp32. I followed the steps, when verifying got this error: 
"Arduino: 1.8.15 (Mac OS X), Board: "Arduino Uno"

sketch_aug31a:29:10: fatal error: uniqueID.h: No such file or directory
 #include "uniqueID.h"
          ^~~~~~~~~~~~
compilation terminated.
exit status 1
uniqueID.h: No such file or directory


This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.


Thank you,

**Reply by Raabitco** [reply]
I am not sure what you mean by using CPU? do you mean computer? please be more specific. if you use computer  why you use Arduino software?
